test/models/operatingsystem_test.rb

```diff
@@ -340,4 +340,11 @@ class OperatingsystemTest < ActiveSupport::TestCase
       assert operatingsystem.save
     end
   end
+
+  describe '#boot_filename' do
+    test 'should be the ipxe unattended url for iPXE' do
+      host = FactoryBot.build(:host, :managed, pxe_loader: 'iPXE')
+      assert_equal 'http://foreman.some.host.fqdn/unattended/iPXE', host.operatingsystem.boot_filename(host)
+    end
+  end
```

Comment:
Can you also add a test with Templates feature on host subnet on? That should render via Smart Proxy. Thanks.